File: Easy/Fibonacci series up to Nth term/fibonacci-series-up-to-nth-term.py
#User function Template for python3

class Solution:
    def series(self, n):
        arr = [0] * (n+1)
        arr[1] = 1
        for i in range(2, len(arr)):
            arr[i] = int((arr[i-1] + arr[i-2]) % (1e9+7))
            
        return arr


#{ 
 # Driver Code Starts
#Initial Template for Python 3

if __name__ == '__main__':
    t = int(input())
    for _ in range(t):
        N = int(input())
        ob = Solution()
        result = ob.series(N)
        print(*result)
# } Driver Code Ends

File: Easy/Fibonacci series up to Nth term/fibonacci-series-up-to-nth-term.java
//{ Driver Code Starts
import java.io.*;
import java.util.*;

class Main {
    // Driver code
    public static void main(String[] args) throws Exception {
        BufferedReader br =
            new BufferedReader(new InputStreamReader(System.in));
        int t = Integer.parseInt(br.readLine().trim());
        while (t-- > 0) {
            int n = Integer.parseInt(br.readLine().trim());
            Solution obj = new Solution();
            int ans[] = obj.Series(n);
            for (int i : ans) {
                System.out.print(i + " ");
            }
            System.out.println();
        }
    }
}
// } Driver Code Ends


class Solution {

    int[] Series(int n) {
        int arr[] = new int[n+1];
        arr[0] = 0;
        arr[1] = 1;
        int MOD = (int)(1e9+7);
        for(int i = 2; i <= n; i++)
            arr[i] = (arr[i-1] % MOD + arr[i-2] % MOD) % MOD;
        return arr;
    }
}